flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/utils/RelTreeWriterImpl.scala

```diff
@@ -147,7 +150,20 @@ class RelTreeWriterImpl(
           }
           val stateTtlHints = FlinkHints.getAllStateTtlHints(rel.asInstanceOf[Hintable].getHints)
           if (stateTtlHints.nonEmpty) {
-            printValues.add(Pair.of("stateTtlHints", RelExplainUtil.hintsToString(stateTtlHints)))
+            // do not merge the state ttl hints for the original plan because we can observe
+            // the processing of state ttl hints in the AST stage
+            val isNeedMerge =
```

Comment:
Can we move this logic to `QueryHintResolver`? I think we don't need to merge twice here.